outputs.tf

```diff
@@ -22,3 +22,24 @@ output "powerbi_sql_dw_server_password" {
   description = "Password of the user dedicated to Power BI"
   value       = contains(random_password.sql_powerbi_viewer, 0) ? random_password.sql_powerbi_viewer[0].result : ""
 }
+
+output "service_principal_client_id" {
+  description = "Client ID of the service principal that is used for service connections"
+  value       = local.application_id
+}
+
+output "service_principal_client_secret" {
```

Comment:
You should not output as it will be in cleartext, i would suggest to use something else, like a secret in kv

Reply:
it is marked as sensitive, so it's not outputted (it's in the state, but it's in clear text in there anyway)
this output is meant to be used by users of the module so they can put it in their own kv
should I create a key vault as part of the module?